File: scripts/algorithms/A/Add One Row to Tree/Add One Row to Tree.py
// Runtime: 77 ms (Top 6.3%) | Memory: 18.50 MB (Top 98.9%)

class Solution:
    def addOneRow(self, root: TreeNode, v: int, d: int) -> TreeNode:
        if d == 1: return TreeNode(v, root, None)
        elif d == 2:
            root.left = TreeNode(v, root.left, None)
            root.right = TreeNode(v, None, root.right)
        else:
            if root.left: self.addOneRow(root.left, v, d-1)
            if root.right: self.addOneRow(root.right, v, d-1)
        return root

File: scripts/algorithms/A/Add One Row to Tree/Add One Row to Tree.java
// Runtime: 0 ms (Top 100.0%) | Memory: 44.40 MB (Top 37.52%)

class Solution {
    public TreeNode addOneRow(TreeNode root, int v, int d) {
        if (d == 1) return new TreeNode(v, root, null);
        else if (d == 2) {
            root.left = new TreeNode(v, root.left, null);
            root.right = new TreeNode(v, null, root.right);
        } else {
            if (root.left != null) addOneRow(root.left, v, d-1);
            if (root.right != null) addOneRow(root.right, v, d-1);
        }
        return root;
    }
}
